tools/crm_resource_print.c

```diff
@@ -382,6 +382,76 @@ resource_check_xml(pcmk__output_t *out, va_list args) {
     return rc;
 }
 
+PCMK__OUTPUT_ARGS("resource-search", "GListPtr", "pe_resource_t *", "const char *")
+static int
+resource_search_default(pcmk__output_t *out, va_list args)
+{
+    GListPtr nodes = va_arg(args, GListPtr);
+    pe_resource_t *rsc = va_arg(args, pe_resource_t *);
+    const char *requested_name = va_arg(args, const char *);
+
+    bool printed = false;
+    int rc = pcmk_rc_no_output;
+
+    if (!out->is_quiet(out) && nodes == NULL) {
+        out->err(out, "resource %s is NOT running", requested_name);
+        return rc;
+    }
+
+    for (GListPtr lpc = nodes; lpc != NULL; lpc = lpc->next) {
+        pe_node_t *node = (pe_node_t *) lpc->data;
+
+        if (!printed) {
+            out->begin_list(out, NULL, NULL, "Nodes");
+            printed = true;
+            rc = pcmk_rc_ok;
+        }
+
+        if (out->is_quiet(out)) {
+            out->list_item(out, "node", "%s", node->details->uname);
+        } else {
+            const char *state = "";
+
+            if (!pe_rsc_is_clone(rsc) && rsc->fns->state(rsc, TRUE) == RSC_ROLE_MASTER) {
+                state = " Master";
+            }
+            out->list_item(out, "node", "resource %s is running on: %s%s",
+                           requested_name, node->details->uname, state);
+        }
+    }
+
+    if (printed) {
+        out->end_list(out);
+    }
+
+    return rc;
+}
+
+
+PCMK__OUTPUT_ARGS("resource-search", "GListPtr", "pe_resource_t *", "const char *")
+static int
+resource_search_xml(pcmk__output_t *out, va_list args)
+{
+    GListPtr nodes = va_arg(args, GListPtr);
+    pe_resource_t *rsc = va_arg(args, pe_resource_t *);
+    const char *requested_name = va_arg(args, const char *);
+
+    xmlNode *xml_node = pcmk__output_xml_create_parent(out, "nodes");
+
+    xmlSetProp(xml_node, (pcmkXmlStr) "resource", (pcmkXmlStr) requested_name);
+
+    for (GListPtr lpc = nodes; lpc != NULL; lpc = lpc->next) {
+        pe_node_t *node = (pe_node_t *) lpc->data;
+        xmlNode *sub_node = pcmk__output_create_xml_text_node(out, "node", node->details->uname);
+
+        if (!pe_rsc_is_clone(rsc) && rsc->fns->state(rsc, TRUE) == RSC_ROLE_MASTER) {
+            xmlSetProp(sub_node, (pcmkXmlStr) "state", (pcmkXmlStr) "master");
```

Comment:
Let's s/master/promoted/ for future proofing

Reply:
Good idea.  I've also adjusted the schema to match.